### ISSUE TITLE
Poetry shells cannot be warpified on mac

### ISSUE THREAD

**Issue by WaelRabah** [warpdotdev/Warp#5399]
### Dupe Check

- [X] I have [searched Warp bugs](https://github.com/warpdotdev/warp/issues?q=is%3Aissue+label%3ABugs) and there are no duplicates

### Describe the bug

The behavior is that I try to warpify a poetry shell. The shell would get stuck and I cannot interact with it in any way.

### To reproduce

1. [Install poetry](https://python-poetry.org/docs/)
2. Create a poetry virtualenv and start a poetry shell: https://python-poetry.org/docs/basic-usage/
3. Warpify the shell
4. Don't panic. Just close the current warp tab and it's gone.

### Expected behavior

After warpification. The shell is no longer responsive to the keyboard.

### Operating system

MacOS

### Operating system and version

mac os sequoia 15.0

### Shell Version

3.2.57

### Current Warp version

v0.2024.10.15.08.02.stable_03

### Regression

No, this bug or issue has existed throughout my experience using Warp

### Does this block you from using Warp daily?

No

### Is this an issue only in Warp?

Yes, I confirmed that this only happens in Warp, not other terminals.

### Warp Internal (ignore): linear-label:b9d78064-c89e-4973-b153-5178a31ee54e

None

**Comment by dannyneira** [comment]
Hi @WaelRabah sorry to hear about this issue. I wasn't able to reproduce this with both the pipx and homebrew versions of poetry. Is there any chance you're using an older version of have some kind of alias or other version of poetry?

run `poetry -V` to check the version
run `which -a poetry` to check for installed versions or aliases

**Comment by WaelRabah** [comment]
Hey @dannyneira, thank you for your response and for looking into this. I apologize for getting back to you so late.

Here are the versions:
```
$ poetry -V
Poetry (version 1.8.3)
```

```
$ which -a poetry
/opt/homebrew/bin/poetry
```

**Comment by andrewginns** [comment]
I also have the same issue, after warpifying a poetry shell I'm unable to interact with it.

Poetry installed via the official installer rather than homebrew or pipx.

```
$ poetry -V
Poetry (version 1.8.3)
```
```
$ which -a poetry
/Users/aginns/.local/bin/poetry
```

**Comment by WaelRabah** [comment]
It was fixed for me in the `v0.2024.11.19.08.02.stable_01` release. Thanks !

**Comment by andrewginns** [comment]
Still having the issue on my setup on the same version.

**Comment by andrewginns** [comment]
Fixed for me on `v0.2024.11.19.08.02.stable_03` release <3
